source/Quantities/Serialization/ScalarBuilder.cs

```diff
@@ -24,25 +24,30 @@ public static IBuilder Create(in QuantityModel model, in UnitRepository repo, in
     }
 
     private static Creator CreateSi<TSi>() where TSi : ISiUnit, IDim => i => i.Inject<Si<TSi>>();
-    private static Creator CreateSiAlias<TSi, TDim>() where TSi : ISiUnit, IDim, IAlias<TDim> where TDim : IDim, ILinear => i => Injector<TSi, TDim>(i).Inject<Si<TSi>>();
+    private static Creator CreateSiAlias<TSi, TDim>() where TSi : ISiUnit, IDim, IAlias<TDim> where TDim : IDim, ILinear => i => AliasInjector<TSi, TDim>(i).Inject<Si<TSi>>();
+    private static Creator CreateSiInvertible<TSi, TDim>() where TSi : ISiUnit, IDim, IInvertible<TDim> where TDim : IDim, ILinear => i => InvertibleInjector<TSi, TDim>(i).Inject<Si<TSi>>();
```

Comment:
Repeating Invertible for all ScalarBuilder seems repetitive. I assume that a general rule, which should invert any ScalarBuilder would come with a readability drawback, but I'm unsure. I'm also thinking about rounding errors, but again, I'm unsure.

Reply:
Valid points. This thing is one of the few classes that relies on reflection where I "chose" which of these methods to bind to through naming convention. Reflection is bad and naming conventions are bad. And it shows here nicely.
Since it's all in one class with reflection targeting private methods within the same class, I can live with it for now. But it'd be a good topic to work on together to find a more sane solution :-)

Reply:
I'd love to discuss this with you at some point.

As a teaser the relevant reflection calls are in the method `GetMethod` (which is an ok name, I promise) and there the local method `GetUnitTypeArgs` on line 64.

Reply:
See also issue #72

Reply:
Yes, I'd like to talk about this :)